### PR TITLE
Issues/7272 update login error screen

### PR DESCRIPTION
**Pull request by aerych**
Refs #7272 

This PR switches the remote error handling from the modal SignupErrorViewController to the new FancyAlertViewController.
The error logic (decision branching) code has been ported more or less intact with a few exceptions:
- methods were renamed to better suit the alert controller. 
- the 405 check for XMLRPC errors has been dropped. Since XML-RPC is enabled by default in versions of WordPress the app supports we shouldn't see this error unless a user chooses to disable XML-RPC via a plugin.  Even if a user were to disable XML-RPC via a plugin the way it was being handled was incorrect.

To test:
To test the implementation, I opted to manually present errors using each option and confirm that the alerts are presented correctly, dismiss correctly and that the more button opens the FAQ, the Support vc, or Helpshift as appropriate. 

Needs review: @nheagy 

simulator screen shot jul 14 2017 4 38 41 pm

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift [hunk 1]
```diff
@@ -47,4 +50,165 @@ extension FancyAlertViewController {
...
+    ///
+    private static func alertForBadURLMessage(_ message: String) -> FancyAlertViewController {
+        let moreHelpButton = ButtonConfig(Strings.moreHelp) { controller in
+            controller.dismiss(animated: true) {
+                // Find the topmost view controller that we can present from
+                guard let appDelegate = UIApplication.shared.delegate,
+                    let window = appDelegate.window,
+                    let viewController = window?.topmostPresentedViewController else { return }
+
+                let url = URL(string: "https://apps.wordpress.org/support/#faq-ios-3")!
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift]: I see that `!`. 

HE'S GOT A GRENADE

File: WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift [hunk 2]
```diff
@@ -47,4 +50,165 @@ extension FancyAlertViewController {
...
+                let webController = WPWebViewController(url: url)
+                let navController = UINavigationController(rootViewController: webController!)
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift]: Another `!`

❗️

File: WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift [hunk 3]
```diff
@@ -47,4 +50,165 @@ extension FancyAlertViewController {
...
+                return alertForGenericErrorMessageWithHelpshiftButton(message, loginFields: loginFields, sourceTag: sourceTag)
+            }
+        }
+
+        if error.domain != WPXMLRPCFaultErrorDomain && error.code != NSURLErrorBadURL {
+            if HelpshiftUtils.isHelpshiftEnabled() {
+                return alertForGenericErrorMessageWithHelpshiftButton(message, loginFields: loginFields, sourceTag: sourceTag)
+
+            } else {
+                return alertForGenericErrorMessage(message, sourceTag: sourceTag)
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift]: You mentioned you had a plan for this `else` clause?

File: WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
-        }
+    struct Strings {
+        static let titleText = NSLocalizedString("What's my site address?", comment: "Title of alert helping users understand their site address")
+        static let bodyText = NSLocalizedString("Your site address appears in the bar at the the top of the screen when you visit your site in Safari.", comment: "Body text of alert helping users understand their site address")
+        static let OK = NSLocalizedString("OK", comment: "Ok button for dismissing alert helping users understand their site address")
+        static let moreHelp = NSLocalizedString("Need more help?", comment: "Title of the more help button on alert helping users understand their site address")
+    }
 
-        typealias Button = FancyAlertViewController.Config.ButtonConfig
+    typealias ButtonConfig = FancyAlertViewController.Config.ButtonConfig
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/FancyAlertViewController+LoginError.swift]: This is definitely a more logical name, but I was just copying what @frosty did.

ITS HIS FAULT.

*points*
*runs away*

File: WordPress/Classes/ViewRelated/NUX/LoginViewController.swift [hunk 5]
```diff
@@ -117,7 +117,11 @@ extension LoginViewController: SigninWPComSyncHandler, LoginFacadeDelegate {
             return
         }
 
-        displayError(error as NSError, sourceTag: sourceTag)
+        let presentingController = navigationController ?? self
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/LoginViewController.swift]: My only real change request: I think this should probably go into `displayError(_ error: NSError, sourceTag: SupportSourceTag)`. This is called from a few places still, like `LoginLinkRequestViewController` ln 108.

I _think_ that'll be a pretty straight forward change tho.

**Comment by aerych** [comment]
Back to ya sir!

**Comment by aerych** [comment]
Thanks @nheagy !!!

**Pull request merged**